File: csharp/unittest-practice/src/main/QuesoChihuahua.cs
﻿namespace unittestpractice.main
{
    internal class QuesoChihuahua : IQueso
    {
        public bool Melted;
        public int Temperature;
        public int Melting = 20;

        public bool IsMelted()
        {
            return Melted;
        }

        public int GetCurrentTemperature()
        {
            return Temperature;
        }

        public int GetMeltingTemperature()
        {
            return Melting;
        }

        public void SetCurrentTemperature(int temp)
        {
            Temperature = temp;
        }

        public void Melt(bool melted)
        {
            Melted = melted;
        }
    }
}

File: java/src/main/java/mx/iteso/ut/QuesoChihuahua.java
package mx.iteso.ut;

/**
 * Clase de Queso Chihuahua.
 */
public class QuesoChihuahua implements Queso {
    /**Número mágico
     * Melting temperature.
     */
    public static final int MELTINGTEMP = 20;
    /**
     * Booleano que indica si está derretido.
     */
    private boolean melted;
    /**
     * Temperatura actual del queso.
     */
    private int temperature;
    /**
     * Temperatura a la que se derrite el queso.
     */
    private int melting = MELTINGTEMP;

    /**
     * @return si el queso está derretido.
     */
    public boolean isMelted() {
        return this.melted;
    }

    /**
     * @return la temperatura actual del queso.
     */
    public int getCurrentTemperature() {
        return this.temperature;
    }

    /**
     * @return la temperatura a la que se derrite el queso.
     */
    public int getMeltingTemperature() {
        return this.melting;
    }

    /**
     * @param temp asigna la temperatura al queso.
     */
    public void setCurrentTemperature(final int temp) {
        this.temperature = temp;
    }

    /**
     * @param melt derrite el queso.
     */
    public void melt(final boolean melt) {
        this.melted = melt;

    }
}
